src/promiseify.js

```diff
@@ -0,0 +1,79 @@
+var callbackFn = (
```

Comment:
Any reason why we vendor this instead of using something like https://www.npmjs.com/package/denodeify?